### ISSUE TITLE
BigQuery parser cannot handle structs within an IF clause

### ISSUE THREAD

**Issue by tonicebrian** [sqlfluff/sqlfluff#1913]
Working with explicit struct arrays is flagged as PSR.

## Expected Behaviour
Given this valid query in BigQuery:

```
SELECT IF(
        TRUE,
        STRUCT('hello' AS greeting, 'world' AS subject),
        STRUCT('hi' AS greeting, 'there' AS subject)
    ) AS salute
FROM (SELECT 1)
```

## Observed Behaviour
The query fails to parse

```
╰─ sqlfluff lint --dialect bigquery deleteme.sql
== [deleteme.sql] FAIL
L:   2 | P:  13 |  PRS | Line 2, Position 13: Found unparsable section: ",\n
                       | STRUCT('hello' AS greeting, 'w..."
All Finished 📜 🎉!
```

## Dialect
Bigquery

## Version
```
╰─ sqlfluff --version                      
sqlfluff, version 0.7.0
╰─ python --version          
Python 3.8.10
```

## Configuration
No configuration

**Comment by keitherskine** [comment]
Another example:
```sql
SELECT
  CASE
    WHEN a.xxx != b.xxx THEN STRUCT(a.xxx AS M, b.xxx AS N)
  END AS xxx
FROM A
JOIN B ON B.id = A.id
```
output:
```
$ sqlfluff lint -v --dialect bigquery zzz.sql
==== sqlfluff ====
sqlfluff:                0.8.1 python:                  3.6.6
implementation:        cpython dialect:              bigquery
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: zzz.sql ] ===

== [zzz.sql] FAIL
L:   2 | P:   3 |  PRS | Line 2, Position 3: Found unparsable section: 'CASE\n
                       | WHEN a.xxx != b.xxx THEN STRUCT...'
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```
